src/cross_section.F90

```diff
@@ -140,169 +133,160 @@ end subroutine calculate_xs
 ! given index in the nuclides array at the energy of the given particle
 !===============================================================================
 
-  subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_mat, i_nuc_mat, &
-       i_log_union, sqrtkT)
+  subroutine calculate_nuclide_xs(i_nuclide, i_sab, E, i_log_union, sqrtkT)
     integer, intent(in) :: i_nuclide ! index into nuclides array
     integer, intent(in) :: i_sab     ! index into sab_tables array
     real(8), intent(in) :: E         ! energy
-    integer, intent(in) :: i_mat     ! index into materials array
-    integer, intent(in) :: i_nuc_mat ! index into nuclides array for a material
     integer, intent(in) :: i_log_union ! index into logarithmic mapping array or
                                        ! material union energy grid
     real(8), intent(in) :: sqrtkT    ! Square root of kT, material dependent
 
     logical :: use_mp ! true if XS can be calculated with windowed multipole
+    integer :: i_temp ! index for temperature
     integer :: i_grid ! index on nuclide energy grid
     integer :: i_low  ! lower logarithmic mapping index
     integer :: i_high ! upper logarithmic mapping index
     real(8) :: f      ! interp factor on nuclide energy grid
+    real(8) :: kT     ! temperature in MeV
     real(8) :: sigT, sigA, sigF ! Intermediate multipole variables
-    type(Nuclide), pointer :: nuc
-    type(Material), pointer :: mat
-
-    ! Set pointer to nuclide and material
-    nuc => nuclides(i_nuclide)
-    mat => materials(i_mat)
-
-    ! Check to see if there is multipole data present at this energy
-    use_mp = .false.
-    if (nuc % mp_present) then
-      if (E >= nuc % multipole % start_E/1.0e6_8 .and. &
-           E <= nuc % multipole % end_E/1.0e6_8) then
-        use_mp = .true.
-      end if
-    end if
 
-    ! Evaluate multipole or interpolate
-    if (use_mp) then
-      ! Call multipole kernel
-      call multipole_eval(nuc % multipole, E, sqrtkT, sigT, sigA, sigF)
-
-      micro_xs(i_nuclide) % total = sigT
-      micro_xs(i_nuclide) % absorption = sigA
-      micro_xs(i_nuclide) % elastic = sigT - sigA
-
-      if (nuc % fissionable) then
-        micro_xs(i_nuclide) % fission = sigF
-        micro_xs(i_nuclide) % nu_fission = sigF * nuc % nu(E, EMISSION_TOTAL)
+    associate (nuc => nuclides(i_nuclide))
+      ! Check to see if there is multipole data present at this energy
+      use_mp = .false.
+      if (nuc % mp_present) then
+        if (E >= nuc % multipole % start_E/1.0e6_8 .and. &
+             E <= nuc % multipole % end_E/1.0e6_8) then
+          use_mp = .true.
+        else
+          ! If using multipole data but outside the RRR, just pick the first
+          ! temperature
+          i_temp = 1
```

Comment:
The problem is that you have to pick something and it's likely that the cell temperature does not match any of the data, so doing a search would come up empty. Once we add interpolation, we could change this to find the bounding temperatures. Also keep in mind this really only matters for the URR. In the fast range, the cross sections will be the same for all temperatures.